src/main/java/org/opencds/cqf/utilities/IOUtils.java

```diff
@@ -406,4 +410,103 @@ public static String formatFileName(String baseName, Encoding encoding) {
         }
         return list;
     }
+
+    public static String getLibraryPathAssociatedWithCqlFileName(String cqlPath) throws FileNotFoundException {
+        String[] split = cqlPath.split("\\\\");
+        String fileName = split[split.length - 1].replaceAll(".cql", ".json");
+        for (String path : IOUtils.getLibraryPaths()) {
+            if(path.contains("library-") && path.contains(fileName)) {
+                return path;
+            }
+        }
+        throw new FileNotFoundException("Could not find a Library Resource Associated with: " + cqlPath);
+    }
+
+    private static HashSet<String> cqlLibraryPaths = new HashSet<String>();
+    public static HashSet<String> getCqlLibraryPaths() {
+        return cqlLibraryPaths;
+    }
+    public static void setupCqlLibraryPaths() {  
```

Comment:
This should not be public and it should trigger when the get is first attempted.

Reply:
And it should be named "initialize".

Reply:
I would have it be null, check for null, and create it the first time if it was null (lazy instantiation).